### PR TITLE
[MRG] replace _TempDir -> tmpdir

### LINKED ISSUE #711: Change tests from _TempDir to tmpdir
Our CI often fails on issues like this one:

> E       FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmp_mne_tempdir_json6b4j3\\participants.json'

see e.g., https://github.com/mne-tools/mne-bids/runs/1924242938?check_suite_focus=true

Should we change from MNE-Python's `_TempDir()` to pytest's `tmpdir`? I think we are doing that in the MNE-Python codebase as well, right? It seems to me like pytest's `tmpdir` is more stable.

Changes would look like this:

```diff
-def test_convert_dataset_format(dir_name, fname, reader):
+def test_convert_dataset_format(tmpdir, dir_name, fname, reader):
    """Test force-converting files to RECOMMENDED formats."""
-   bids_root = _TempDir()
+   bids_root = tmpdir
```

### PR DESCRIPTION
**Pull request by sappelhoff**
closes #711 

~WIP, gonna do this piece by piece. Please tell me early if you think this is bad.~

Time to run tests:

- master:  Results (228.05s)
- this PR: Results (214.55s)

But the main point is to ditch an MNE private func for a pytest inbuilt function that is more versatile and that more people know about/use.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [x] PR title starts with [MRG]
- ~[whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated~
- [x] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Review by agramfort — APPROVED** [review]
LGTM

File: mne_bids/tests/test_copyfiles.py [hunk 1]
```diff
@@ -50,27 +49,26 @@ def test_get_brainvision_encoding():
...
-    test_dir = _TempDir()
     data_path = op.join(base_path, 'brainvision', 'tests', 'data')
     raw_fname = op.join(data_path, 'test.vhdr')
 
     with pytest.raises(ValueError):
         _get_brainvision_paths(op.join(data_path, 'test.eeg'))
 
     # Write some temporary test files
-    with open(op.join(test_dir, 'test1.vhdr'), 'w') as f:
+    with open(tmpdir / 'test1.vhdr', 'w') as f:
```
Comment (hoechenberger) [inline mne_bids/tests/test_copyfiles.py:61]: You can actually use the OOP approach offered through Path here

```suggestion
    with (tmpdir / 'test1.vhdr').open('w') as f:
```

But I guess it's a matter of taste

Comment (sappelhoff) [reply mne_bids/tests/test_copyfiles.py:61]: good to know :+1: happy to accept the suggestion if there is a benefit. As of taste, I am not sure what I like better.

Comment (hoechenberger) [reply mne_bids/tests/test_copyfiles.py:61]: Let's just leave it the wait it currently is then :)

**Review by hoechenberger — APPROVED** [review]

**Comment by hoechenberger** [comment]
Thanks @sappelhoff!

**Pull request merged**